src/main/cli/cli.c

```diff
@@ -4808,7 +4826,7 @@ if (buildKey) {
                 cliPrint("), ");
             }
         }
-        if (!gpsIsHealthy()) {
+        if (gpsData.state < GPS_STATE_RECEIVING_DATA) {
```

Comment:
Agree, done

Reply:
If it's `GPS_STATE_CONFIGURE` (without a "D" at the end) then it's fine 👍

Reply:
Hmm.
For MSP connected GPS, we bypass CONFIGURE, so we can only be in RECEIVING_DATA or higher.
But I notice that LOST_COMMUNICATION is above RECEIVING_DATA.
Perhaps the best test is `(gpsData.state = GPS_STATE_CONFIGURE)`?

Reply:
My suggestion doesn't change anything in the algorithm. It needs to be `gpsData.state <= GPS_STATE_CONFIGURE` in order to be equal to `gpsData.state < GPS_STATE_RECEIVING_DATA`

Reply:
Agree, however maybe what we really need is `gpsData.state == GPS_STATE_RECEIVING_DATA`, because otherwise we would show 'connected' when we are in `GPS_STATE_LOST_COMMUNICATION`

Reply:
I think best just leave it as is.

Reply:
My suggestion leaves everything as is. It is just easier to see that we check all states **before** `GPS_STATE_CONFIGURE` and then once check all states **after** `GPS_STATE_CONFIGURE`.

In the current situation you always need to go back to the definition of `gpsState_e` enum to check what actually happens.

Reply:
Ultimately, an external module, such as cli.c, should not be looking into the internal state data of gps.c. I think that's a bad dependency.

The status command in cli is asking the questions:
Is the GPS connected?
Is the GPS configured?

I think the GPS module should provide query functions to answer those questions and decouple how the question is answered from the external module. It might still check the state at this point, but if the states get refactored, cli doesn't break or care.

I added a review comment about this earlier, but ctzsnooze and I discussed and thought that might be a requirement for the next refactoring PR rather than this one?

Reply:
We can probably provide enough information for the user  using only `gpsIsHealthy`.

However, the current code has the virtue of differentiating between:
- successfully connecting at the configured baud rate (CLI shows connected, and baud rate)
- successfully configuring the firmware (shows configured as well as the above)

However for any meaningful analysis of problem events, reviewing the debug fields in Sensors provides far more clarity than what we get in CLI.  I think this will be more useful down the track.  It would be great if we could provide a 'decoded' version of the values in the debug fields to explain what they mean, directly in Configurator..

Reply:
OK @ Karatebrot I made it as you suggest, since the code is equivalent, and you are very persistent, despite personally a preference to leave it, I have accepted your changes.